src/Interface.jl

```diff
@@ -0,0 +1,46 @@
+
+
+## Some function which make the model easier to define.
+
+
+# A very light weight container for a state space model
+# The state space is one dimensional!
+# Note that this is only for a very simple demonstration.
+mutable struct Container
+    x::Vector{Float64}
+    num_produce::Float64
+end
+
+# This is important for initalizaiton
+function init()
+    return current_trace()
+end
+
+# Th
+function report_observation(trace, logp::Float64)
```

Comment:
Maybe rename the function to something that's more self-explanatory? Additionally, I guess it would be nice to stick to the convention of adding an exclamation mark at the end of function names if functions modify the arguments in-place. I'm not sure if specifying the type of `logp` is useful here, since you're not dispatching on the type you could just remove it.

Reply:
I am sorry, i did not mention that the interface is only for debugging and demonstration. Maybe i should remove it completely out of this version.